WordPress/src/main/java/org/wordpress/android/ui/accounts/SignInFragment.java

```diff
@@ -1005,12 +1007,105 @@ protected void signIn() {
             if (isUsernameEmail()) {
                 startProgress(getActivity().getString(R.string.checking_email));
                 requestWPComEmailCheck();
+            } else if (isWPComDomain(mUsername)) {
+                // If a wpcom domain was entered, check if the subdomain matches an existing username.
+                String maybeUsername = extractWPComSubDomain(mUsername);
+                if (maybeUsername.length() > 0) {
+                    // See if the username exists.
+                    startProgress(getActivity().getString(R.string.checking_username));
+                    requestWPComUsernameCheck(maybeUsername);
+                } else {
+                    // The text that was entered was .wordpress.com or the like.
+                    // Its invalid so just show an error.
+                    showUsernameError(R.string.username_invalid);
+                }
             } else {
                 showPasswordFieldAndFocus();
             }
         }
     }
 
+    /**
+     * Tests the specified string to see if it contains a wpcom subdomain.
+     *
+     * @param string The string to check
+     * @return True if the string contains a wpcom subdomain, else false.
+     */
+    private boolean isWPComDomain(String string) {
+        Matcher matcher = WPCOM_DOMAIN.matcher(string);
+        return matcher.find();
+    }
+
+    /**
+     * Extracts the subdomain from a wpcom domain string.
+     * @param domain A wpcom domain is expected.
+     * @return A wpcom subdomain or an empty string.
+     */
+    private String extractWPComSubDomain(String domain) {
+        String str = UrlUtils.addUrlSchemeIfNeeded(domain, false);
+        String host = UrlUtils.getHost(str);
+        if (host.length() > 0) {
+            String[] parts = host.split("\\.");
+            if (parts.length > 0) {
+                return parts[0];
+            }
+        }
+        return "";
+    }
+
+    /**
+     * Performs an API request to see if the specified username exists.
+     * If a username does exist, the username field is updated and the password field is displayed.
+     *
+     * @param name The username to check.
+     */
+    private void requestWPComUsernameCheck(String name) {
+        final String username = name;
```

Comment:
Rather than create a new string variable, we can simply use `private void requestWPComUsernameCheck(final String name)`